autodeconstruct.lua

```diff
@@ -257,13 +246,23 @@ function autodeconstruct.build_pipes(drill)
       pipeType = "se-space-pipe"
     end
   end
+  
+  -- Connection position index is different from entity.direction
+  local conn_index = 1  -- north
```

Comment:
[defines.direction is an int typ](https://wiki.factorio.com/Types/Direction)
since there seems to be no [define](https://lua-api.factorio.com/latest/defines.html#defines.direction) for conn_direction I would simplify this using math to something like ` math.floor(drillData.direction/2)+1` and add a short comment reference to the direction defines prototype page

Reply:
Hmm I did that at first, wasn't sure I liked assuming that those constants never change and that it always is even. But I guess it is impossible to rotate any miner 45 degrees.

Reply:
your current code would still break in a 45° rotation, similar to mine, if would just break slightly different

Reply:
Mine would default to North. With divide by 2 in the index, it would crash on a fractional index unless round() is used to get to closest cardinal direction.

Reply:
[math.ceil , math.floor](http://lua-users.org/wiki/MathLibraryTutorial)

Return the integer no greater than or no less than the given value